### PR TITLE
WMRB on Serial Predictions

### PR DESCRIPTION
**Pull request by jfkirk**
Increases efficiency of WMRB (or any sample-based loss) by allowing it to function on a serial prediction of sampled items, rather than sampling from a dense prediction of all items.

### CODE REVIEW

File: tensorrec/tensorrec.py [hunk 1]
```diff
@@ -245,15 +245,21 @@ def _build_tf_graph(self, n_user_features, n_item_features):
...
         self.tf_item_feature_values = tf.placeholder('float', [None])
         self.tf_interaction_indices = tf.placeholder('int64', [None, 2])
         self.tf_interaction_values = tf.placeholder('float', [None])
+
+        self.tf_sample_indices = tf.placeholder('int64', [None, None])
         self.tf_learning_rate = tf.placeholder('float', None)
         self.tf_alpha = tf.placeholder('float', None)
-        self.tf_sampled_item_indices = tf.placeholder('int64', [None, None])
+
+        # from nose.tools import set_trace;set_trace()
```
Comment (jfkirk) [inline tensorrec/tensorrec.py]: note: remove

File: tensorrec/tensorrec.py [hunk 2]
```diff
@@ -339,18 +361,23 @@ def _build_tf_graph(self, n_user_features, n_item_features):
...
         if self.loss_graph_factory.is_dense:
             loss_graph_kwargs.update({
                 'tf_prediction': self.tf_prediction,
-                'tf_interactions': tf_interactions,
                 'tf_rankings': self.tf_rankings,
             })
         if self.loss_graph_factory.is_sample_based:
-            tf_sample_predictions = gather_sampled_item_predictions(
-                tf_prediction=self.tf_prediction, tf_sampled_item_indices=self.tf_sampled_item_indices
+            tf_sample_predictions = densify_sampled_item_predictions(
```
Comment (jfkirk) [inline tensorrec/tensorrec.py]: Note: should rename one of these `tf_sample_predictions`
